### DISCUSSION TITLE
Visual Scripting for Signal Connections

### DISCUSSION THREAD

**Discussion by ParadineDev** [godotengine/godot-proposals#9090] [Editor]
Signals allow you to pull links between logic components through the editor rather than hardcoding them. But when there are many logic components and connections, it is quite difficult to understand and edit such connections in the editor.

It would be very convenient to have visual programming that has only two functions - send a signal and receive a signal.

**Comment by ParadineDev** [comment]
It could look like this:

![image](https://github.com/godotengine/godot-proposals/assets/119674972/a874af8a-aa88-4a36-b021-520491da328d)

**Comment by luckyabsoluter** [comment]
Adding API annotations that serve as the role of the public keyword in other languages and marking them to be connected to signals first would be convenient.
For Example, @export_func and @api.
